domain/schema/secret.go

```diff
@@ -309,25 +309,86 @@ CREATE TABLE
 CREATE UNIQUE INDEX idx_secret_role_role ON secret_role (role);
 
 INSERT INTO secret_role VALUES
-    (0, 'view'),
-    (1, 'rotate'),
+    (0, 'none'),
+    (1, 'view'),
     (2, 'manage');
 
--- TODO: probably we need a model level permission table like the
--- recently added "permission" table in the controller database.
+CREATE TABLE
+    secret_grant_subject_type (
+        id   INT PRIMARY KEY,
+        type TEXT
+    );
+
+INSERT INTO secret_grant_subject_type VALUES
+    (0, 'unit'),
+    (1, 'application'),
+    (2, 'model');
+
+CREATE TABLE
+    secret_grant_scope_type (
+        id   INT PRIMARY KEY,
+        type TEXT
+    );
+
+INSERT INTO secret_grant_scope_type VALUES
+    (0, 'unit'),
+    (1, 'application'),
+    (2, 'model'),
+    (3, 'relation');
+
 CREATE TABLE
     secret_permission (
```

Comment:
Please undo this odd indentation here and in the other secrets DDL.

Reply:
Done in commit https://github.com/juju/juju/pull/17212/commits/c589cbfa51ca0c38d265e5410e135f99c06bac97